### PR TITLE
fix: add back custom sql filtering with Query as source

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Re-enabling template processing for Query object inside `ExploreMixin` class. This was missed in the initial implementation, but discovered in the last release. This PR also fixes issues with saving adhoc custom sql when using a `Query` to be saved when creating a `SqlaTable`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by jinghua-qa** [comment]
I found an issue that if user set a filter use custom-sql filter, after save the chart,  the custom sql fiter will be removed.

https://user-images.githubusercontent.com/81597121/186702056-3e682d60-3e42-4007-8636-d8359b5788cb.mov

File: superset-frontend/src/explore/controlUtils/getControlValuesCompatibleWithDatasource.ts [hunk 1]
```diff
@@ -23,8 +23,10 @@ import {
   isAdhocMetricSimple,
   isSavedMetric,
   isSimpleAdhocFilter,
+  isFreeFormAdhocFilter,
   JsonValue,
   SimpleAdhocFilter,
+  FreeFormAdhocFilter,
```
Comment (eschutho) [inline superset-frontend/src/explore/controlUtils/getControlValuesCompatibleWithDatasource.ts:29]: do you still need this import for `FreeFormAdhocFilter`?

Comment (hughhhh) [reply superset-frontend/src/explore/controlUtils/getControlValuesCompatibleWithDatasource.ts]: nope removed it

**Review by eschutho — APPROVED** [review]

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Pull request merged**